plugin/pkg/shoot/validator/admission.go

```diff
@@ -1766,23 +1771,60 @@ func getDefaultMachineImage(machineImages []gardencorev1beta1.MachineImage, imag
 			}
 		}
 		if defaultImage == nil {
-			return nil, field.Invalid(fldPath, imageName, fmt.Sprintf("no valid machine image found that support architecture `%s`", *arch))
+			return nil, field.Invalid(fldPath, imageReference, fmt.Sprintf("no valid machine image found that supports architecture `%s`", *arch))
+		}
+	}
+
+	var (
+		machineImageVersionMajor *uint64
+		machineImageVersionMinor *uint64
+	)
+
+	if image != nil {
+		versionParts := strings.Split(strings.TrimPrefix(image.Version, "v"), ".")
+		if len(versionParts) == 3 {
+			return image, nil
+		}
+		if len(versionParts) == 2 && len(versionParts[1]) > 0 {
+			v, err := strconv.ParseUint(versionParts[1], 10, 0)
+			if err != nil {
+				return nil, field.Invalid(fldPath, image.Version, fmt.Sprintf("must be a semantic version `%s`", *arch))
```

Comment:
Did you mean to format the invalid, semantic image version into the string instead of the architecture?
```suggestion
				return nil, field.Invalid(fldPath, image.Version, fmt.Sprintf("must be a semantic version `%s`", *image.Version))
```